pkg/controllers/disruption/drift_test.go

```diff
@@ -623,6 +630,28 @@ var _ = Describe("Drift", func() {
 			Expect(ExpectNodes(ctx, env.Client)).To(HaveLen(1))
 			ExpectExists(ctx, env.Client, nodeClaim)
 		})
+		It("should not ignore nodes with a terminationGracePeriod that have pods with the karpenter.sh/do-not-disrupt annotation", func() {
```

Comment:
I noticed that the drift/expiration/consolidtion test cases didn't have any explicit tests for blocking PDBs. My thinking is the drift tests are only verifying the drift vs consolidation logic is working as expected, which doesn't involve do-not-disrupt vs blocking PDBs. The disruption controller suite_test.go does have test cases for verifying the behavior of do-not-disrupt vs blocking PDBs.

Reply:
I've ensured that all combinations of eventual vs graceful disruption, TGP vs no TGP, and blocking PDBs vs do-not-disrupt annotation are covered in the test cases in the disruption controller `suite_test.go`. I believe that adding additional cases to each disruption type handler would be duplicative and unnecessary. Please unresolve this thread if you feel that it would be beneficial to add additional testing.